### PR TITLE
[CI] [GHA] Increase job timeouts

### PR DESCRIPTION
**Pull request by akashchi**
### Tickets:
 - *124037*

### CODE REVIEW

File: .github/workflows/windows.yml [hunk 1]
```diff
@@ -188,7 +188,7 @@ jobs:
 
   Samples:
     needs: Build
-    timeout-minutes: 10
+    timeout-minutes: 20
```
Comment (akashchi) [inline .github/workflows/windows.yml:191]: https://github.com/openvinotoolkit/openvino/actions/runs/6704508115/job/18218972510

**Review by ilya-lavrenov — APPROVED** [review]
let's also port to releases/2023/2 branch

File: .github/workflows/android_arm64.yml [hunk 2]
```diff
@@ -27,7 +27,7 @@ concurrency:
 
 jobs:
   Build:
-    timeout-minutes: 15
+    timeout-minutes: 40
```
Comment (ilya-lavrenov) [inline .github/workflows/android_arm64.yml:30]: does it assume that build without cache can also pass here?
I mean the worst case when everything needs to be rebuilt.

Comment (akashchi) [reply .github/workflows/android_arm64.yml:30]: I am unsure. I found the longest one with 1h of build time: https://github.com/openvinotoolkit/openvino/actions/runs/6623899432/job/17991799138. Strangely enough, the `cache-hit-rate` is ~97% as indicated by logs:
image

Comment (akashchi) [reply .github/workflows/android_arm64.yml:30]: The recent pipelines are in around the 10-minute mark (when they do not try to exceed the 15-minute timeout).

**Pull request merged**